oai_agents/common/multi_setup_trainer.py

```diff
@@ -0,0 +1,166 @@
+import concurrent.futures
+import random
+from scripts.utils.common import generate_name
+from oai_agents.common.learner import LearnerType
+from oai_agents.common.tags import Prefix, KeyCheckpoints
+from oai_agents.agents.rl import RLAgentTrainer
+import dill
+
+
+class MultiSetupTrainer:
+    def __init__(
+            self,
+            args,
+            train_types,
+            eval_types,
+            curriculum,
+            tag_for_returning_agent
+        ):
+        self.args = args
+        self.train_types = train_types
+        self.eval_types = eval_types
+        self.curriculum = curriculum
+        self.tag_for_returning_agent = tag_for_returning_agent
+
+        self.parallel = args.parallel
+        self.num_of_training_variants = args.num_of_training_variants
+        self.for_evaluation = args.for_evaluation
+
+    def generate_hdim_and_seed(self):
+        training_seeds = [1010, 2020, 2602, 13, 68, 2907, 105, 128]
+        training_hdims = [256] * len(training_seeds)
+
+        evaluation_seeds = [3031, 4041, 5051, 3708, 3809, 3910, 4607, 5506]
+        evaluation_hdims = [256] * len(evaluation_seeds)
+
+        if self.for_evaluation:
+            seeds = evaluation_seeds
+            hdims = evaluation_hdims
+            min_seed, max_seed = 3000, 5999
+        else:
+            seeds = training_seeds
+            hdims = training_hdims
+            min_seed, max_seed = 0, 2999
+
+        selected_seeds = []
+        selected_hdims = []
+
+        if self.num_of_training_variants <= len(seeds):
+            selected_seeds = seeds[:self.num_of_training_variants]
+            selected_hdims = hdims[:self.num_of_training_variants]
+        else:
+            selected_seeds = seeds[:]
+            selected_hdims = hdims[:]
+
+            remaining = self.num_of_training_variants - len(seeds)
+            available_seeds = list(set(range(min_seed, max_seed + 1)) - set(selected_seeds))
+            random_seeds = random.sample(available_seeds, remaining)
```

Comment:
why do we reuse a seed if there aren't enough seeds in training_seeds? Shouldn't we just make sure that training_seeds and evaluation_seeds always have enough unique seeds?

Reply:
We did not reuse seeds here. 

We firstly have 8 unique seeds for both training and evaluation. However, if we need more, we sample some other seeds from 0 to 2999 for training and some other from 3000 to 5999 for evaluation.

I am unsure whether I answered your concerns. If not, feel free to let me know. Thanks!

Reply:
I thought about that. But if we let user to input additional seeds, that may not be easy to use. Do you have any suggestion changes for us to input additional seeds? (If we just let them to use arguments, I think that will make the code confused to understand. We may need other ways to implement.)

Reply:
I propose just hard coding enough seeds so we don't have to randomly sample new seed. This way, the seeds stay the same during each run, but if we randomly sample seeds, we won't get similar seed. 

I also agree with Joe, that we should throw an error if the user requires more seed. In that case, the user should just come and update the function with more seeds (it doesn't have to be an additional argument)

Reply:
I think 8 seeds is enough for our project and I will throw an error message to ask them to modify seeds setting in which function. Thank you both for suggestions.